straxen/plugins/peaks/peak_se_score.py

```diff
@@ -0,0 +1,149 @@
+        mask = peaks["type"] == 2
+        mask &= (peaks["area"] > self.se_selection_area_roi[0]) & (
+            peaks["area"] < self.se_selection_area_roi[1]
+        )
+        mask &= (peaks["range_50p_area"] > self.se_selection_width_roi[0]) & (
+            peaks["range_50p_area"] < self.se_selection_width_roi[1]
+        )
+        mask &= peaks["area_fraction_top"] != 0
+        return mask
+
+    @staticmethod
+    @njit
+    def get_se_count_and_pdf_sum(peaks, se_peaks, se_indices, para_a, para_b, eps):
+        se_nearby_probability = np.zeros(len(peaks))
+        for index, peak_i in enumerate(peaks):
+            indices = se_indices[index]
+            se_in_time = se_peaks[indices[0] : indices[1]]
+            peak_area_top = peak_i["area"] * (peak_i["area_fraction_top"] + eps)
+            se_area_top = se_in_time["area"] * (se_in_time["area_fraction_top"] + eps)
+            peak_position_resolution = para_a / np.sqrt(peak_area_top) + para_b
+            se_position_resolution = para_a / np.sqrt(se_area_top) + para_b
+            combined_position_resolution = np.sqrt(
+                se_position_resolution**2 + peak_position_resolution**2
+            )
+            d_squre = (se_in_time["x"] - peak_i["x"]) ** 2 + (se_in_time["y"] - peak_i["y"]) ** 2
+            constant = 1 / (2 * np.pi * combined_position_resolution**2)
+            exponent = np.exp(-1 / 2 * (d_squre / combined_position_resolution**2))
+            _se_nearby_prob = constant * exponent
+            se_nearby_probability[index] = np.sum(_se_nearby_prob)
+        return se_nearby_probability
+
+    def compute_se_score(self, peaks, _peaks):
+        """Function to calculate the SE score for each peak."""
+        # select single electrons
+        mask = self.select_se(peaks)
+        se_peaks = peaks[mask]
+        # get single electron indices in peak vicinity
+        split_peaks = np.zeros(len(_peaks), dtype=strax.time_fields)
+        split_peaks["time"] = _peaks["center_time"] - self.se_time_search_window_left
+        split_peaks["endtime"] = _peaks["center_time"] + self.se_time_search_window_right
+        split_result = strax.touching_windows(se_peaks, split_peaks)
+        # get se score
+        # eps: smallest positive float, used to prevent division by zero.
+        eps = np.finfo(float).eps
+        _se_nearby_probability = self.get_se_count_and_pdf_sum(
+            _peaks,
+            se_peaks,
+            split_result,
+            self._para_a,
+            self._para_b,
+            eps,
+        )
+        return _se_nearby_probability
+
+    def compute(self, peaks):
+        # sort peaks by center_time
+        argsort = np.argsort(peaks["center_time"], kind="mergesort")
+        _peaks = np.sort(peaks, order="center_time")
+        mask_nan = np.isnan(_peaks["x"]) | np.isnan(_peaks["y"])
+        # prepare output
+        se_nearby_probability = np.zeros(len(peaks))
+        _se_nearby_probability = np.zeros(len(peaks))
```

Comment:
Minor one, if you could, `np.zeros(len(peaks))` -> `np.full(len(peaks), np.nan)`. Then you do not need `_se_nearby_probability[mask_nan] = np.nan`.